frontend-react/src/components/MessageTracker/MessageReceivers.tsx

```diff
@@ -1,62 +1,511 @@
+                    activeColumnSortOrder !== FilterOptionsEnum.NONE,
+            })}
+            onClick={handleColHeaderClick}
+            rowSpan={rowSpan}
+        >
+            <div className="column-header-title-container">
+                <p
+                    className={classnames({
+                        "text-pre":
+                            columnHeaderTitle === ColumnDataEnum.FileName,
+                    })}
+                >
+                    {columnHeaderTitle}
+                </p>
+                {sortIcon}
+            </div>
+        </th>
+    );
+};
+
+const MessageReceiversRow = ({
+    receiver,
+    activeColumn,
+    activeColumnSortOrder,
+    setModalText,
+    modalRef,
+}: MessageReceiversRowProps) => {
+    // We highlight the entire active column so within the row,
+    // we need to know which column is active
+    const checkForActiveColumn = (colName: string) =>
+        classnames({
+            "active-col-td":
+                colName === activeColumn &&
+                activeColumnSortOrder !== FilterOptionsEnum.NONE,
+        });
+
+    const uppercaseMain = receiver.Main.toUpperCase();
+    return (
+        <tr>
+            <td className={checkForActiveColumn(ColumnDataEnum.Name)}>
+                {receiver.Name}
+            </td>
+            <td className={checkForActiveColumn(ColumnDataEnum.Service)}>
+                {receiver.Service}
+            </td>
+            <td className={checkForActiveColumn(ColumnDataEnum.Date)}>
+                {receiver.Date}
+            </td>
+            <td
+                role="button"
+                className={`message-receiver-break-word ${checkForActiveColumn(
+                    ColumnDataEnum.ReportId
+                )}`}
+                onClick={() => {
+                    setModalText({
+                        title: `${ColumnDataEnum.ReportId}:`,
+                        body: receiver.ReportId,
+                    });
+                    modalRef?.current?.toggleModal();
+                }}
+            >
+                {receiver.ReportId}
+            </td>
+            <td className={checkForActiveColumn(ColumnDataEnum.Main)}>
+                <p
+                    className={classnames(
+                        "font-mono-sm border-1px bg-primary-lighter radius-md padding-left-1 padding-right-1 margin-top-0",
+                        {
+                            "bg-blue-5": uppercaseMain === StatusEnum.BATCH,
+                            "bg-blue-10": uppercaseMain === StatusEnum.PROCESS,
+                            "bg-blue-20": uppercaseMain === StatusEnum.READY,
+                        }
+                    )}
+                >
+                    {uppercaseMain}
+                </p>
+            </td>
+            <td className={checkForActiveColumn(ColumnDataEnum.Sub)}>
+                {receiver.Sub}
+            </td>
+            <td
+                role="button"
```

Comment:
Oh!  One issue with going with `role="button"` is that we'll have to update `tabIndex="0"` for keyboard navigation.  This is handled by default by interactive elements like buttons and links, so we should use those first if possible.

Reply:
Excellent call out! Added.